bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/RequestValidationService.kt

```diff
@@ -172,58 +174,87 @@ class RequestValidationService(
+        val identifiersValidator = ValidateIdentifierTypesExists(mainAddressRequests, SiteCreateError.MainAddressIdentifierNotFound)
+        val identifiersDuplicateValidator = ValidateAddressIdentifiersDuplicated(mainAddressRequests, SiteCreateError.MainAddressDuplicateIdentifier)
 
         return requests.flatMap { request ->
             val mainAddress = request.site.mainAddress
-
-            val validationErrors = validateRegionExists(mainAddress, addressMetadata.regions, SiteCreateError.MainAddressRegionNotFound, request) +
-                    validateIdentifierTypesExists(mainAddress, addressMetadata.idTypes, SiteCreateError.MainAddressIdentifierNotFound, request) +
-                    validateParentBpnExists(request.bpnlParent, request.index, existingParentBpns, SiteCreateError.LegalEntityNotFound) +
-                    validateAddressIdentifiersDuplicated(
-                        address = mainAddress,
-                        existingIdentifiers = addressDuplicateIdentifierCandidates,
-                        bpn = null,
-                        error = SiteCreateError.MainAddressDuplicateIdentifier,
-                        entityKey = request
-                    )
+            val validationErrors = regionValidator.validate(mainAddress, request) +
+                    identifiersValidator.validate(mainAddress, request) +
+                    identifiersDuplicateValidator.validate(mainAddress, request, null)
 
             validationErrors.map { Pair(request, it) }
-
         }.groupBy({ it.first }, { it.second })
     }
 
     fun validateSiteUpdates(
         requests: Collection<SitePartnerUpdateRequest>,
     ): Map<SitePartnerUpdateRequest, Collection<ErrorInfo<SiteUpdateError>>> {
 
-        val mainAddressRequests = requests.map { it.site.mainAddress }
-        val addressMetadata = metadataService.getMetadata(mainAddressRequests).toKeys()
+        val requestByLeError = validateSiteUpdatesInternal(requests)
+        val errorsByRequestAddress = validateSiteAddressUpdates(requests)
+        return mergeErrorMaps(requestByLeError, errorsByRequestAddress)
+    }
+
+    fun validateSiteUpdatesInternal(
+        requests: Collection<SitePartnerUpdateRequest>,
+    ): Map<SitePartnerUpdateRequest, Collection<ErrorInfo<SiteUpdateError>>> {
 
         val requestedSiteBpns = requests.map { it.bpns }
         val existingSiteBpns = siteRepository.findDistinctByBpnIn(requestedSiteBpns).map { it.bpn }.toSet()
+        val existingBpnValidator = ValidateUpdateBpnExists(existingSiteBpns, SiteUpdateError.SiteNotFound)
 
-        val addressDuplicateIdentifierCandidates = getAddressDuplicateIdentifierCandidates(mainAddressRequests)
+        return requests.flatMap { request ->
+            val validationErrors = existingBpnValidator.validate(request.bpns)
+
+            validationErrors.map { Pair(request, it) }
+
+        }.groupBy({ it.first }, { it.second })
```

Comment:
You could simplify the whole block to
```suggestion
        return requests.associateWith { request ->
            existingBpnValidator.validate(request.bpns)
        }
```

There are several cases in this class that could be replaced similarly where currently 
`.flatMap{ ... }.groupBy({ it.first }, { it.second })`
is used.

Reply:
This is already done in the next implementation step. 
After the associateWith  a  `.filterValues { it.isNotEmpty() } ` is needed